test/replace_allocate.cpp

```diff
@@ -0,0 +1,155 @@
+#include <migraphx/allocation_model.hpp>
+#include <migraphx/replace_allocate.hpp>
+#include <migraphx/dead_code_elimination.hpp>
+#include <migraphx/pass_manager.hpp>
+#include <migraphx/check_shapes.hpp>
+#include <migraphx/argument.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/make_op.hpp>
+#include <migraphx/stringutils.hpp>
+#include <migraphx/register_op.hpp>
+#include <basic_ops.hpp>
+#include <test.hpp>
+
+struct cpu_allocate : migraphx::auto_register_op<cpu_allocate>
+{
+    migraphx::shape s{};
+
+    template <class Self, class F>
+    static auto reflect(Self& self, F f)
+    {
+        return migraphx::pack(f(self.s, "shape"));
+    }
+
+    std::string name() const { return "cpu::allocate"; }
```

Comment:
Dont use the same names for operators to avoid any potential conflict with names.

Reply:
Will it make a difference here since the cpu/hip allocates are not registered as ops because we didn't link those libraries in?

Reply:
We could link them in the future. It also makes sure we dont rely on the name of the operator in replace_allocate. Either way, I dont think we need different allocation operators for the tests. At most we could have different allocation_models(although I am not sure if thats necessary either).